### ISSUE TITLE
something is wrong with resnext-50 pretrained model 

### ISSUE THREAD

**Issue by kaishijeng** [terrychenism/ResNeXt#3]
Terry

I use the predict.py below with cat.jpg and output of resnext-50 model is total wrong.
But resnet-50 output is correct to predict it as tabby cat.
DO you know what could be the issue?

Thanks

Resnext-50:

probability=0.627792, class=n01729322 hognose snake, puff adder, sand viper
probability=0.151374, class=n02091244 Ibizan hound, Ibizan Podenco
probability=0.125211, class=n01514668 cock
probability=0.027394, class=n02097047 miniature schnauzer
probability=0.006769, class=n02097474 Tibetan terrier, chrysanthemum dog

Resnet-50:

probability=0.213489, class=n02123045 tabby, tabby cat
probability=0.156405, class=n02119022 red fox, Vulpes vulpes
probability=0.127035, class=n02123159 tiger cat
probability=0.115639, class=n02119789 kit fox, Vulpes macrotis
probability=0.092057, class=n02124075 Egyptian cat

predict.py:

import os, urllib
def download(url):
    filename = url.split("/")[-1]
    if not os.path.exists(filename):
        urllib.urlretrieve(url, filename)
def get_model(prefix, epoch):
    download(prefix+'-symbol.json')
    download(prefix+'-%04d.params' % (epoch,))

#get_model('http://data.mxnet.io/models/imagenet/resnet/50-layers/resnet-50', 0)
#get_model('http://data.mxnet.io/models/imagenet/resnext/50-layers/resnext-50', 0)
#get_model('http://data.mxnet.io/models/imagenet/resnext/50-layers/resnext-101', 0)
import mxnet as mx
#sym, arg_params, aux_params = mx.model.load_checkpoint('resnet-50', 0)
#sym, arg_params, aux_params = mx.model.load_checkpoint('resnext-50', 0)
sym, arg_params, aux_params = mx.model.load_checkpoint('resnext-101', 0)
mod = mx.mod.Module(symbol=sym, context=mx.gpu())
mod.bind(for_training = False,
         data_shapes=[('data', (1,3,224,224))])
mod.set_params(arg_params, aux_params)
#download('http://data.mxnet.io/models/imagenet/resnet/synset.txt')
with open('synset.txt') as f:
    synsets = [l.rstrip() for l in f]
import tarfile
download('http://data.mxnet.io/data/val_1000.tar')
tfile = tarfile.open('val_1000.tar')
tfile.extractall()
with open('val_1000/label') as f:
    val_label = [int(l.split('\t')[0]) for l in f]
import numpy as np
import cv2
def get_image(filename):
    img = cv2.imread(filename)  # read image in b,g,r order
    img = cv2.cvtColor(img, cv2.COLOR_BGR2RGB)   # change to r,g,b order
    img = cv2.resize(img, (224, 224))  # resize to 224*224 to fit model
    img = np.swapaxes(img, 0, 2)
    img = np.swapaxes(img, 1, 2)  # change to (channel, height, width)
    img = img[np.newaxis, :]  # extend to (example, channel, heigth, width)
    return img
from collections import namedtuple
Batch = namedtuple('Batch', ['data'])
img = get_image('./cat.jpg')
mod.forward(Batch([mx.nd.array(img)]))
prob = mod.get_outputs()[0].asnumpy()
prob = np.squeeze(prob)
y = np.argsort(prob)[::-1]
for i in y[0:5]:
   print('probability=%f, class=%s' %(prob[i], synsets[i]))

**Comment by terrychenism** [comment]
resnext-50 use different label, plz download from: http://data.dmlc.ml/mxnet/models/imagenet/resnext/

**Comment by kaishijeng** [comment]
It works,

Thanks.

**Comment by beniz** [comment]
@terrychenism hi the link above is broken, any other ways to grab the labels for the pre-trained models ? Thanks.

**Comment by beniz** [comment]
OK, got it from the DMLC inception_v3 1K tarball, and so in the meantime I've put it here: https://deepdetect.com/models/resnext/synset.txt
